server/routerlicious/packages/routerlicious-base/src/nexus/app.ts

```diff
@@ -0,0 +1,88 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { TypedEventEmitter } from "@fluidframework/common-utils";
+import { ICollaborationSessionEvents } from "@fluidframework/server-lambdas";
+import { json, urlencoded } from "body-parser";
+import compression from "compression";
+import cookieParser from "cookie-parser";
+import express from "express";
+import { Provider } from "nconf";
+import { DriverVersionHeaderName } from "@fluidframework/server-services-client";
+import {
+	alternativeMorganLoggerMiddleware,
+	bindCorrelationId,
+	bindTelemetryContext,
+	bindTimeoutContext,
+	jsonMorganLoggerMiddleware,
+} from "@fluidframework/server-services-utils";
+import { RestLessServer, IHttpServerConfig } from "@fluidframework/server-services";
+import { BaseTelemetryProperties, HttpProperties } from "@fluidframework/server-services-telemetry";
+import { catch404, getIdFromRequest, getTenantIdFromRequest, handleError } from "../utils";
+
+export function create(
+	config: Provider,
+	collaborationSessionEventEmitter?: TypedEventEmitter<ICollaborationSessionEvents>,
+) {
+	// Maximum REST request size
+	const requestSize = config.get("nexus:restJsonSize");
+	const httpServerConfig: IHttpServerConfig = config.get("system:httpServer");
+	// Express app configuration
+	const app: express.Express = express();
```

Comment:
Why are we creating an Express app for a service that only handles socket connections? This file should be entirely unnecessary

Reply:
This is a good point. From what I understand, from SocketIO perspective, it makes no difference if you are using Express or running standalone. I do see the case that it could be slightly more resource-efficient to not have to setup Express. And it should have  no negative impact in SocketIO.

However for now, while we still have Alfred and changes on it being developed it might be good to keep Nexus tracking most of the Alfred code? Once we remove Alfred, and there is nothing that we have to keep importing and patching to Nexus, we can also remove Express. 

Like it sounds, it might not be necessary for this PR and it can be saved for PR#2 where we remove Alfred websocket? Not sure, I'm not familiar enough with FRS deployment cycles to make the call. Like if features take a long time to be fully rolled out, it might be best to just test it now and do the change in this PR.

Reply:
That's fair. My opinion is that we should remove it now so that no one comes along and adds an HTTP endpoint into Nexus since the design is not meant to accommodate HTTP APIs. I think you're right that the Express App will not add too much draw on resources (probably just a slight boot time penalty), but from a code cleanliness and design enforcement perspective, it seems counter-intuitive to have all this server code.

At the very least, I would recommend removing all but the most-barebones aspects of the Express server so that it doesn't get used. Like

```ts
const app: express.Express = express();
app.use((req, res) => {
  res.json({ message: "OK" });
});
return app;
```

Reply:
Yes, I agree on the maintainability aspect. I'll just go ahead and remove it as it sounds like it might get difficult to get back to it later on.